bids/layout/layout.py

```diff
@@ -36,6 +36,26 @@ def commonpath(paths):
 
 __all__ = ['BIDSLayout']
 
+MANDATORY_BIDS_FIELDS = {
+    "Name": {"Name": "Example dataset"},
+    "BIDSVersion": {"BIDSVersion": "1.0.2"},
+}
+
+MANDATORY_DERIVATIVES_FIELDS = {
+    **MANDATORY_BIDS_FIELDS,
+    "PipelineDescription.Name": {"PipelineDescription": {"Name": "Example pipeline"}},
+}
+
+EXAMPLE_BIDS_DESCRIPTION = {
+    inner_key: inner_value for outer_key in MANDATORY_BIDS_FIELDS.keys()
+    for inner_key, inner_value in MANDATORY_BIDS_FIELDS[outer_key].items()
+}
+
+EXAMPLE_DERIVATIVES_DESCRIPTION = {
+    inner_key: inner_value for outer_key in MANDATORY_DERIVATIVES_FIELDS.keys()
+    for inner_key, inner_value in MANDATORY_DERIVATIVES_FIELDS[outer_key].items()
+}
```

Comment:
I *think* this is the same:

```suggestion
EXAMPLE_BIDS_DESCRIPTION = {**val for val in MANDATORY_BIDS_FIELDS.values()}

EXAMPLE_DERIVATIVES_DESCRIPTION = {**val for val in MANDATORY_DERIVATIVES_FIELDS.values()}
```

Reply:
That's also what I wanted to work, but I get this error when trying the unpacking strategy:
```
{**val for val in MANDATORY_BIDS_FIELDS.values()}
  File "<stdin>", line 1
SyntaxError: dict unpacking cannot be used in dict comprehension
```

but let me know if you prefer any other solutions or one of the solutions [outlined on stack overflow](https://stackoverflow.com/questions/37584544/dict-merge-in-a-dict-comprehension)

Reply:
Ah sorry. Well, we can at least be a bit briefer:

```suggestion
EXAMPLE_BIDS_DESCRIPTION = {
    k: val[k] for val in MANDATORY_BIDS_FIELDS.values() for k in val}

EXAMPLE_DERIVATIVES_DESCRIPTION = {
    k: val[k] for val in MANDATORY_DERIVATIVES_FIELDS.values() for k in val}
```

Reply:
thanks! your suggestion is cleaner.